api/proto/teleport/legacy/client/proto/authservice.proto

```diff
@@ -1832,10 +1832,11 @@ message UpstreamInventoryHello {
   // its auth token. i.e. Services is the subset of SystemRoles that are currently
   // active.
   repeated string Services = 3 [(gogoproto.casttype) = "github.com/gravitational/teleport/api/types.SystemRole"];
-
-  // TODO(fspmarshall): look into what other info can safely be stated here once, instead of
-  // being repeatedly announced (e.g. addrs, static labels, etc). may be able to achieve a
-  // non-trivial reduction in network usage by doing this.
+  // Hostname is the hostname associated with the instance. This value is not required or guaranteed
+  // to be unique and its validity is not enforceable (i.e. join tokens do not constrian what an
+  // instance can claim its hostname to be). This value exists only to assist users in correlating
+  // instance resources with hosts.
+  string Hostname = 4;
```

Comment:
This is the same `Hostname` value that all the services heartbeat, so I think it's best to keep the name consistent.  These comments are mostly here to hedge against future confusion.  Since `UpstreamInventoryHello` is the main source of truth used to decide the capabilities of the control stream (e.g. `Services` are checked against the callers cert, and then used to decide what it can heartbeat), I think it merits being very explicit about the quality of the information it contains.